### PR TITLE
react-tabs: Updated to use react-theme tokens and typography

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Removed tab.constants.ts that was defining missing tokens and typography constants
- Update tab styles and animation to use  tokens from react-theme

## Issues

#Fixes 22254

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f65f187799f50793c9263cb299d0426a28cc4e9a'>f65f187799f50793c9263cb299d0426a28cc4e9a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/Avatar-mount_base.html">1057</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/Avatar-mount.html">1036</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/Button-mount_base.html">654</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/Button-mount.html">636</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProvider-mount_base.html">2238</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProvider-mount.html">2195</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProviderWithTheme-mount_base.html">333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProviderWithTheme-mount.html">342</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">280</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">286</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">361</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">367</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/MakeStyles-mount_base.html">1818</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22703/perf-test-react-components/MakeStyles-mount.html">1818</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts [hunk 1]
```diff
@@ -43,23 +37,23 @@ const useRootStyles = makeStyles({
...
   mediumHorizontal: {
-    columnGap: pendingSpacingTokens.sNudge,
+    columnGap: tokens.spacingHorizontalSNudge,
     justifyContent: 'center',
-    ...shorthands.padding(pendingSpacingTokens.m, pendingSpacingTokens.mNudge),
+    ...shorthands.padding(tokens.spacingVerticalM, tokens.spacingHorizontalMNudge),
   },
   mediumVertical: {
-    columnGap: pendingSpacingTokens.sNudge,
+    columnGap: tokens.spacingHorizontalSNudge,
```
Comment (miroslavstastny) [inline packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts:45]:
```suggestion
    columnGap: tokens.spacingVerticalSNudge,
```

There are perhaps more places where vertical should be used

**Comment by GeoffCoxMSFT** [comment]
react-tabs component moved into the react-components folder, so have to create a new PR.

**Pull request closed without merging**